### PR TITLE
Single competition query

### LINKED ISSUE #32: Single Competition GraphQL Query
- [ ] Create graphql query to return information for a single track from the database
- [ ] create type for track query

**request**
query {
  track(id: ID) {
    id
    trackPath
    month
    year
    verses { verseType }
  }
}﻿

**response**
{
  "data": {
    "track": {
      "id": <string>,
      "trackPath": <string>,
      "month": <integer>,
      "year": <integer>
      "verses": [ { verseType } ] <- array of verse objects
    }
  }
}

### PR DESCRIPTION
**Pull request by the-color-bliu**
## Create Query to Get Single Competition
Create CompetitionType, add to query type, create test

This PR closes #32

- [ ] Bug fix (non-breaking change which fixes an issue): 

- [x] New feature (non-breaking change which adds functionality):

- [ ] Breaking change (fix or feature that would cause existing functionality to change):

- [ ] My change requires a change to the documentation.

- [ ] I have updated the documentation accordingly.

- [x] I have added tests to cover my changes.

- [x] All new and existing tests passed.

### CODE REVIEW

**Review by ETBassist — APPROVED** [review]
LGTM

File: spec/factories.rb [hunk 1]
```diff
@@ -9,6 +9,9 @@
     track_path { Faker::Internet.url }
     month { rand(13) }
     year { 2021 }
+    description { Faker::Games::Pokemon.name }
```
Comment (SageOfCode) [inline spec/factories.rb:12]: Love the Pokemon theme.

File: app/graphql/types/competition_type.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module Types
+  class CompetitionType < Types::BaseObject
+    field :id, ID, null: false
+    field :track_path, String, null: false
+    field :month, Integer, null: false
+    field :year, Integer, null: false
+    field :description, String, null: true
```
Comment (SageOfCode) [inline app/graphql/types/competition_type.rb:7]: Does this field default to true?

**Review by nmking22 — APPROVED** [review]

**Review by SageOfCode — APPROVED** [review]
Awesome, this looks ready to merge.

**Pull request merged**